src/helpers/charts/chart_common.ts

```diff
@@ -429,3 +440,119 @@ export function getBaselineArrowDirection(
+  const rect = getters.getVisibleRect(getters.getActiveMainViewport());
+
+  const scrollableViewportWidth = Math.min(rect.width, width - offsetCorrectionX);
+  const scrollableViewportHeight = Math.min(rect.height, height - offsetCorrectionY);
+
+  const position = {
+    x: offsetCorrectionX + offsetX + Math.max(0, (scrollableViewportWidth - chartSize.width) / 2),
+    y: offsetCorrectionY + offsetY + Math.max(0, (scrollableViewportHeight - chartSize.height) / 2),
+  }; // Position at the center of the scrollable viewport
+
+  return position;
+}
+
+/**
+ * Return a "smart" chart definition in the given zone. The definition is "smart" because it will
+ * use the best type of chart to display the data of the zone.
+ *
+ * It will also try to find labels and datasets in the range, and try to find title for the datasets.
+ *
+ * The type of chart will be :
+ * - If the zone is a single empty cell, returns a bar chart
+ * - If the zone is a single non-empty cell, returns a scorecard
+ * - If the all the labels are numbers/date, returns a line chart
+ * - Else returns a bar chart
+ */
+export function getSmartChartDefinition(zone: Zone, getters: Getters): ChartDefinition {
+  let dataSetZone = zone;
+  if (zone.left !== zone.right) {
+    dataSetZone = { ...zone, left: zone.left + 1 };
+  }
+  const dataSets = [zoneToXc(dataSetZone)];
+  const sheetId = getters.getActiveSheetId();
+
+  const topLeftCell = getters.getCell(sheetId, zone.left, zone.top);
+  if (getZoneArea(zone) === 1 && topLeftCell?.content) {
+    return {
+      type: "scorecard",
+      title: "",
+      background: BACKGROUND_CHART_COLOR,
+      keyValue: zoneToXc(zone),
+      baselineMode: DEFAULT_SCORECARD_BASELINE_MODE,
+      baselineColorUp: DEFAULT_SCORECARD_BASELINE_COLOR_UP,
+      baselineColorDown: DEFAULT_SCORECARD_BASELINE_COLOR_DOWN,
+    };
+  }
+
+  let title = "";
+  const cells = getters.getCellsInZone(sheetId, {
+    ...dataSetZone,
+    bottom: dataSetZone.top,
+  });
+  const dataSetsHaveTitle = !!cells.find(
+    (cell) => cell && cell.evaluated.type !== CellValueType.number
+  );
+
+  if (dataSetsHaveTitle) {
+    const texts = cells.reduce((acc, cell) => {
+      const text = cell && cell.evaluated.type !== CellValueType.error && getters.getCellText(cell);
+      if (text) {
+        acc.push(text);
+      }
+      return acc;
+    }, [] as string[]);
+
+    const lastElement = texts.splice(-1)[0];
+    title = texts.join(", ");
+    if (lastElement) {
+      title += (title ? " " + _t("and") + " " : "") + lastElement;
+    }
+  }
+
+  let labelRangeXc: string | undefined;
+  if (zone.left !== zone.right) {
+    labelRangeXc = zoneToXc({
+      ...zone,
+      right: zone.left,
+      top: dataSetsHaveTitle ? zone.top + 1 : zone.top,
+    });
+  }
+  const newLegendPos = dataSetZone.right === dataSetZone.left ? "none" : "top"; //Using the same variable as above to identify number of columns involved.
```

Comment:
the comment looks a bit "franglish"
it's either useless or could be replaced by something like "Only display legend for several datasets"
and could put it above the line of code because it's prettier :p

Reply:
I have to admit, I have no clue what the original comment tried to say...